GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2282,7 +2335,8 @@ private void generateListOfChangesInSubmodulesChangesToolStripMenuItem_Click(obj
                 .ToDictionary(localPath =>
                 {
                     var submodule = configFile.ConfigSections.FirstOrDefault(section => section.GetValue("path").Trim() == localPath);
-                    return submodule?.SubSection?.Trim();
+                    Assumes.NotNull(submodule?.SubSection);
+                    return submodule.SubSection.Trim();
```

Comment:
This lambda is returning the key for the dictionary. Null keys are not allowed. Swaps one kind of exception for another, but is more explicit about the issue.